### PR TITLE
Dynamic authentication

### PR DESCRIPTION
**Pull request by jlopezmalla**
## What changes were proposed in this pull request?

Change Vault temp tokens for authentication for Vault's appRole authentication.

THIS PR HAD TO BE MERGER AFETER #11 

## How was this patch tested?

tested in a DC/OS 1.8.8 cluster with calico.

### CODE REVIEW

File: mesos/src/main/scala/org/apache/spark/deploy/rest/mesos/MesosRestServer.scala [hunk 1]
```diff
@@ -104,12 +107,14 @@ private[mesos] class MesosSubmitRequestServlet(
...
+    val sparkPropertiesToSend = if (sparkProperties.get("spark.secret.vault.role").isDefined) {
       if (sparkProperties.isDefinedAt("spark.secret.vault.host")) {
         val vaultUrl = sparkProperties("spark.secret.vault.host")
-        val vaultToken = sparkProperties("spark.secret.vault.token")
+        val role = sparkProperties("spark.secret.vault.role")
+        val vaultToken = VaultHelper.getTokenFromAppRole(vaultUrl,
+          appRoleToken, role)
         val temporalToken = VaultHelper.getTemporalToken(vaultUrl, vaultToken)
-        logDebug(s"Vault Temp token $temporalToken")
+        logDebug(s"login in vault using app: $temporalToken and role: $role")
```
Comment (arinconstrio) [inline mesos/src/main/scala/org/apache/spark/deploy/rest/mesos/MesosRestServer.scala]: Why not Trace instead debug??

File: core/src/main/scala/org/apache/spark/security/VaultHelper.scala [hunk 2]
```diff
@@ -29,16 +29,51 @@ import org.apache.spark.internal.Logging
 
 object VaultHelper extends Logging {
 
-  lazy val jsonTemplate: String = "{ \"token\" : \"_replace_\" }"
+  lazy val jsonTempTokenTemplate: String = "{ \"token\" : \"_replace_\" }"
```
Comment (arinconstrio) [inline core/src/main/scala/org/apache/spark/security/VaultHelper.scala]: Suggestion, extract all the pure http logic to and compose the object, in order to have unit test over the vault logic.

File: mesos/src/main/scala/org/apache/spark/deploy/rest/mesos/MesosRestServer.scala (hunk 1 above)
Comment (jlopezmalla) [inline mesos/src/main/scala/org/apache/spark/deploy/rest/mesos/MesosRestServer.scala]: Ok Trace will be more accurate I will change it

**Comment by Gschiavon** [comment]
LGTM

File: core/src/main/scala/org/apache/spark/security/VaultHelper.scala (hunk 2 above)
Comment (jlopezmalla) [inline core/src/main/scala/org/apache/spark/security/VaultHelper.scala]: Okay, but, maybe this change has to be made in another PR?

**Comment by jlopezmalla** [comment]
changes made @arinconstrio  and a question added.
